### PR TITLE
Update README.md

### LINKED ISSUE #45: Establish and document nomenclature for data model release versions
Discussed during 24-1 sprint kick-off on 2024.01.05:. Possible approaches:

- v[year].[month].[release number] - this version is used for HTAN data model releases, and is also used by FAIR for schematic and DCA releases.

- OR, take software approach of designating major vs minor patches (major.minor.patches-date) - e.g. 1.1.0-202401 . This approach would be more intuitive for devs and is complemented by the release notes including information on date of release.

Group agreed to adopt latter software approach. Action required to **define and document** what constitutes major vs minor vs patches, and how this nomenclature should be adopted for the MC2 data model.

Consider also standard approach for what is included in release notes. in addition to relevant issues/PR incorporated into each release, would be helpful to also document which version of schematic and DCA the released data model is level with (at the time of release)

@aditigopalan and @jaybee84 feel free to discuss this further in comments about how we define these and where this should be documented.

### PR DESCRIPTION
**Pull request by aditigopalan**
Adding release notes and versioning information to the README

Fixes #45

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,3 +35,7 @@ When a new valid value needs to be added to the data model:
 
 4. When ready to release a new version of the data model and a pull request is made, the valid values will automatically get added to `mc2.model.csv` and its JSON-LD file.
 
+## Release Notes 
+
+The latest version release is v23.12.1. Moving forward, this repository will adopt <a href="https://semvar.org/" target="_blank"> Semanting Versioning</a>  (SemVar), which provides a standard way of versioning software based on meaningful changes in functionality.
```
Comment (jaybee84) [inline README.md]:
```suggestion
The version release tag for January 2024 is v23.12.1. Moving forward, this repository will adopt <a href="https://semvar.org/" target="_blank"> Semanting Versioning</a>  (SemVar), which provides a standard way of versioning software based on meaningful changes in functionality.
```

Comment (jaybee84) [reply README.md]: I added a small edit. Also, the link here is not going to the right place. Could you please check the link?

**Review by jaybee84 — CHANGES_REQUESTED** [review]
@aditigopalan added a couple of comments, please take a look.

**Comment by aditigopalan** [comment]
@jaybee84 done!

**Review by jaybee84 — APPROVED** [review]

**Pull request merged**